packages/liveblocks-react/src/liveblocks.tsx

```diff
@@ -193,55 +255,46 @@ export function createLiveblocksContext<
     return;
   }
 
-  function useInboxNotificationsSelectorCallback(
-    state: CacheState<BaseMetadata>
-  ): InboxNotificationsState {
-    const query = state.queries[INBOX_NOTIFICATIONS_QUERY];
+  let inboxNotificationsSubscribers = 0;
 
-    if (query === undefined || query.isLoading) {
-      return {
-        isLoading: true,
-      };
-    }
+  function useSubscribeToInboxNotificationsEffect(options?: {
+    autoFetch: boolean;
+  }) {
+    const autoFetch = useInitial(options?.autoFetch ?? true);
+    useEffect(() => {
+      if (autoFetch) {
+        void fetchInboxNotifications();
+      }
 
-    if (query.error !== undefined) {
-      return {
-        error: query.error,
-        isLoading: false,
-      };
-    }
+      // Increment
+      inboxNotificationsSubscribers++;
+      poller.start(POLLING_INTERVAL);
 
-    return {
-      inboxNotifications: selectedInboxNotifications(state),
-      isLoading: false,
-    };
+      return () => {
+        // Decrement
+        if (inboxNotificationsSubscribers <= 0) {
+          console.warn(
+            `Internal unexpected behavior. Cannot decrease subscriber count for query "${INBOX_NOTIFICATIONS_QUERY}"`
+          );
+          return;
+        }
+
+        inboxNotificationsSubscribers--;
+        if (inboxNotificationsSubscribers <= 0) {
+          poller.stop();
+        }
+      };
+    }, [autoFetch]);
   }
 
   function useInboxNotifications(): InboxNotificationsState {
-    useEffect(() => {
-      void fetchInboxNotifications();
-      incrementInboxNotificationsSubscribers();
-
-      return () => decrementInboxNotificationsSubscribers();
-    }, []);
```

Comment:
This block is now abstracted as a custom hook `useSubscribeToInboxNotificationsEffect()`. There is no direct access to the inc/dec functions anymore.